share/revng/test/tests/analysis/Decompilation/clang/dual-case-switch.c.model.yml

```diff
@@ -0,0 +1,33 @@
+---
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+Functions:
+  - OriginalName: f
+    Prototype: "/Types/1883-CABIFunctionType"
+  - OriginalName: main
+    Prototype: "/Types/1884-CABIFunctionType"
+Types:
+  - ID: 256
+    Kind: PrimitiveType
+    PrimitiveKind: Void
+    Size: 0
+  - ID: 520
+    Kind: PrimitiveType
+    PrimitiveKind: Generic
+    Size: 8
+  - ID: 1883
+    Kind: CABIFunctionType
+    ABI: SystemV_x86_64
+    ReturnType:
+      UnqualifiedType: "/Types/520-PrimitiveType"
+    Arguments:
+      - Index: 0
+        Type:
+          UnqualifiedType: "/Types/520-PrimitiveType"
+  - ID: 1884
+    Kind: CABIFunctionType
+    ABI: SystemV_x86_64
+    ReturnType:
+      UnqualifiedType: "/Types/256-PrimitiveType"
+    Arguments: []
```

Comment:
Is it really important to check the model here? Isn't the `FileCheck` enough? For instance, look at what we do for `pretty-ints`. In that case we only check the architecture, because the checks on the model aren't really important in that case. In principle I'm fine with having an empty YAML if we don't really care about checking the model. This makes the test more robust against changes in the model that are not significant for the feature being tested here. The only problem is that apparently right now `revng model compare` doesn't accept an empty YAML, so for all the tests where the model is not really important we just pass a YAML with only the architecture, which is basically guaranteed to pass unless something really bad happens.

Reply:
I agree with the fact that checking the model here is not particularly significant for the objectives of the test.

The point is that I saw that the current pipeline for the `revng-c.decompilation` tests type check again a known model with `revng model compare` as you mentioned, and I didn't want to craft another identical pipeline with the except of this `revng model compare` check, and at that point I thought of also including the checks against the names of the functions. But I agree that only including the check for the architecture may be more sensible. I'll dot it

Reply:
Ok, I left the architecture check only in https://github.com/revng/revng-c/commit/37b1450b7bb07320407437f8adc5b9739ccb10a6, for what concerns the model check.